partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -360,13 +364,70 @@ public data class TimeType(
     }
 }
 
-public data class TimestampType(override val metas: Map<String, Any> = mapOf()) : SingleType() {
-    override val allTypes: List<StaticType>
-        get() = listOf(this)
+public class TimestampType internal constructor(internal val timestampType: PartiQLTimestampType = PartiQLTimestampType()) : SingleType() {
+
+    override val metas: Map<String, Any> = timestampType.metas
+    override val allTypes: List<StaticType> = listOf(this)
+    @PartiQLTimestampExperimental
+    public val precision: Int? = timestampType.precision
```

Comment:
Null precision means arbitrary precision as defined in Ion I'm assuming?

Reply:
Yes. Will add to kdoc.